backend/apps/ifc_validation/checks/header_policy/validate_header.py

```diff
@@ -90,36 +90,73 @@ class HeaderStructure(ConfiguredBaseModel):
     
     company_name: str = Field(default="")
     application_name: str = Field(default="")
+    schema_identifier: str = Field(default="")
     version: str = Field(default="")
+    mvd: str = Field(default="")
+    ifc_application_info: dict = Field(default_factory=dict)
     
     
     @model_validator(mode='before')
     def populate_header(cls, values):
         if (file := values.get('file')):
-            header = file.wrapped_data.header
+
+            purepythonparser = values.get('purepythonparser')
+            header = file.wrapped_data.header if not purepythonparser else file.header
             file_description = header.file_description
             file_name = header.file_name
             
+            fields = [
+                ('description', 0),
+                ('implementation_level', 1),
+                ('name', 0),
+                ('time_stamp', 1),
+                ('author', 2),
+                ('organization', 3),
+                ('preprocessor_version', 4),
+                ('originating_system', 5),
+                ('authorization', 6)
+            ]
+
```

Comment:
```suggestion
            fields = [
                (file_description, 'description', 0),
                (file_description, 'implementation_level', 1),
                (file_name, 'name', 0),
                (file_name, 'time_stamp', 1),
                (file_name, 'author', 2),
                (file_name, 'organization', 3),
                (file_name, 'preprocessor_version', 4),
                (file_name, 'originating_system', 5),
                (file_name, 'authorization', 6)
            ]

```

If you already do it so elegantly with dictionary comprehensions, I would do it like this. Because the `[:2]` `[2:]` slices are not really that obvious.

Also, maybe an even better approach would be to implement this in the simple_spf parser so that the code base here can be fully generic. But that's probably too late now, maybe as a clean-up action later if you ever feel like it.

Reply:
I was already satisfied but your solution is even more elegant. Thanks :-) 
Implemented it https://github.com/buildingSMART/validate/commit/7a8e80e98589a0396b23423c7d244780041b64fc

> But that's probably too late now, maybe as a clean-up action later if you ever feel like it.
I considered that, but thought it would take too long to get a PR to the step_file_parser, a new ifcopenshell build etc. 
But it's definitely nice for a clean up